src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java

```diff
@@ -53,16 +55,16 @@ protected final void onStartup() {
 
     @SuppressWarnings("unchecked")
     private void initializeDrivingFeatures() {
-        final GATKPath drivingPath = getDrivingFeaturePath();
-        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(drivingPath.toPath());
+        final GATKPath drivingFeaturesPath = getDrivingFeaturePath();
+        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(IOUtils.getPath(drivingFeaturesPath.toString()));
         if (isAcceptableFeatureType(codec.getFeatureType())) {
-            drivingFeatures = new FeatureDataSource<>(new FeatureInput<>(drivingPath), FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null, cloudPrefetchBuffer, cloudIndexPrefetchBuffer, referenceArguments.getReferencePath());
-
-            final FeatureInput<F> drivingFeaturesInput = new FeatureInput<>(drivingPath, "drivingFeatureFile");
+            final FeatureInput<F> drivingFeaturesInput = new FeatureInput<>(drivingFeaturesPath.toString());
+            drivingFeatures = new FeatureDataSource<>(drivingFeaturesInput, FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null, cloudPrefetchBuffer, cloudIndexPrefetchBuffer, referenceArguments.getReferencePath());
             features.addToFeatureSources(0, drivingFeaturesInput, codec.getFeatureType(), cloudPrefetchBuffer, cloudIndexPrefetchBuffer,
                                          referenceArguments.getReferencePath());
+            header = getHeaderForFeatures(drivingFeaturesInput);
         } else {
-            throw new UserException("File " + drivingPath.getRawInputString() + " contains features of the wrong type.");
+            throw new UserException("File " + drivingFeaturesPath.toPath() + " contains features of the wrong type.");
```

Comment:
@mwalker174 Some of these changes appear to revert changes that we made recently. I'm wondering if these are really necessary. Can they be removed (other than the header caching anyway) ? Does something fail without these ?

Reply:
@cmnbroad Thanks for catching this, I think these are errors from merge conflicts. I've reverted most of the changes, but I don't see another way to retrieve the header.